packages/core/src/codewhisperer/models/errors.ts

```diff
@@ -30,8 +30,8 @@ export class ProjectSizeExceededError extends SecurityScanError {
 }
 
 export class DefaultError extends SecurityScanError {
-    constructor() {
-        super('Security scan failed.', 'DefaultError', DefaultCodeScanErrorMessage)
+    constructor(error: string | undefined) {
+        super(`Security scan failed. ${error}`, 'DefaultError', DefaultCodeScanErrorMessage)
```

Comment:
why not pass the error as the `cause` field? also consider `ToolkitError.chain()` .

strinigifying errors randomly in messages completely defeats the builtin mechanisms in the extension such as : https://github.com/aws/aws-toolkit-vscode/blob/6300d520ac347b7596d98c7cec312b67f98ee528/packages/core/src/shared/errors.ts#L467

and makes it much harder to show useful errors to users, in reporting, etc. Please, use care when designing errors.

Reply:
These errors are not shown to the user. This is for telemetry purpose only.

Reply:
This is redundant change. Reverted acc in draft CR

Reply:
> These errors are not shown to the user

This cannot be assumed. Codepaths can change. Errors should use best practices.